featuretools/synthesis/deep_feature_synthesis.py

```diff
@@ -1260,19 +1260,16 @@ def check_primitive(
     return primitive
 
 
-def _all_direct_and_same_path(input_features):
-    return all(
-        isinstance(f, DirectFeature) for f in input_features
-    ) and _features_have_same_path(input_features)
-
-
-def _features_have_same_path(input_features):
+def _all_direct_and_same_path(input_features) -> bool:
```

Comment:
If were slowly adding typing to this repo can you also type out `input_features` as well?

Reply:
Done! I'm not super confident in the signature though...

Reply:
EDIT: I just typed it as list. If we type it as a list of FeatureBase, there are issues with circular imports.

Reply:
Top of the file:
```
from typing import List, TYPE_CHECKING

if TYPE_CHECKING:
     from featuretools import FeatureBase
```

`def _all_direct_and_same_path(input_features: List["FeatureBase"]) -> bool:`

Reply:
thanks!

Reply:
looks like this gets flagged by codecov unfortunately